adam_core/backend/pyoorb.py

```diff
@@ -0,0 +1,326 @@
+                warnings.warn(
+                    f"Invalid argument {k} passed to PYOORB backend. Ignoring..."
+                )
+
+        # If an allowed kwarg is missing, add the default
+        for k in allowed_kwargs:
+            if k not in kwargs:
+                kwargs[k] = PYOORB_CONFIG[k]
+
+        self.dynamical_model = kwargs["dynamical_model"]
+        self.ephemeris_file = kwargs["ephemeris_file"]
+
+        env_var = "ADAM_CORE_PYOORB"
+        if env_var in os.environ.keys() and os.environ[env_var] == "True":
+            pass
+        else:
+            if os.environ.get("OORB_DATA") is None:
+                os.environ["OORB_DATA"] = os.path.join(
+                    os.environ["CONDA_PREFIX"], "share/openorb"
+                )
+
+            # Prepare pyoorb
+            ephfile = os.path.join(os.getenv("OORB_DATA"), self.ephemeris_file)
+            err = oo.pyoorb.oorb_init(ephfile)
+            if err == 0:
+                os.environ[env_var] = "True"
+            else:
+                warnings.warn(f"PYOORB returned error code: {err}")
+
+        return
+
+    @staticmethod
+    def _configure_orbits(
+        orbits: np.ndarray,
+        t0: np.ndarray,
+        orbit_type: str,
+        time_scale: str,
+        magnitude: Optional[float] = None,
+        slope: Optional[float] = None,
+    ) -> np.ndarray:
+        """
+        Convert an array of orbits into the format expected by PYOORB.
+
+        Parameters
+        ----------
+        orbits : `~numpy.ndarray` (N, 6)
+            Orbits to convert. See orbit_type for expected input format.
+        t0 : `~numpy.ndarray` (N)
+            Epoch in MJD at which the orbits are defined.
+        orbit_type : {'cartesian', 'keplerian', 'cometary'}, optional
+            Orbital element representation of the provided orbits.
+            If cartesian:
+                x : heliocentric ecliptic J2000 x position in AU
+                y : heliocentric ecliptic J2000 y position in AU
+                z : heliocentric ecliptic J2000 z position in AU
+                vx : heliocentric ecliptic J2000 x velocity in AU per day
+                vy : heliocentric ecliptic J2000 y velocity in AU per day
+                vz : heliocentric ecliptic J2000 z velocity in AU per day
+            If keplerian:
+                a : semi-major axis in AU
+                e : eccentricity in degrees
+                i : inclination in degrees
+                Omega : longitude of the ascending node in degrees
+                omega : argument of periapsis in degrees
+                M0 : mean anomaly in degrees
+            If cometary:
+                p : perihelion distance in AU
+                e : eccentricity in degrees
+                i : inclination in degrees
+                Omega : longitude of the ascending node in degrees
+                omega : argument of periapsis in degrees
+                T0 : time of perihelion passage in MJD
```

Comment:
What's the advantage of being able to pass in the keplerian and cometary representations to openorb, instead of always using cartesian?

Reply:
Ah so this comes from the olden days before I decided to go full cartesian. That said, we know openorb does not treat all representations equally so I've kept this in the event we ever want to try different representations. Happy to remove it if you want me to.